ext/curl/share.c

```diff
@@ -134,6 +136,143 @@ PHP_FUNCTION(curl_share_strerror)
 }
 /* }}} */
 
+/**
+ * Initialize a persistent curl share handle with the given share options, reusing an existing one if found.
+ *
+ * Throws an exception if the share options are invalid.
+ */
+PHP_FUNCTION(curl_share_init_persistent)
+{
+	zval *share_opts = NULL, *entry = NULL;
+	zend_ulong persistent_id = 0;
+
+	php_curlsh *sh = NULL;
+
+	CURLSHcode error = 0;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_ARRAY(share_opts)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (zend_hash_num_elements(Z_ARRVAL_P(share_opts)) == 0) {
+		zend_argument_value_error(1, "must not be empty");
+		goto error;
+	}
+
+	ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(share_opts), entry) {
+		ZVAL_DEREF(entry);
+
+		bool       failed = false;
+		zend_ulong option = zval_try_get_long(entry, &failed);
+
+		if (failed) {
+			zend_argument_type_error(1, "must contain only integer values, %s given", zend_zval_value_name(entry));
```

Comment:
Not sure how an error message regarding integers should look like with types
```suggestion
			zend_argument_type_error(1, "must contain only int values, %s given", zend_zval_value_name(entry));
```